RELEASING.md

```diff
@@ -1,34 +1,74 @@
 # Releasing
 
-## Go SDK
+> **Warning**
+> Currently, if any SDK bumps an engine dependency, we should bump the engine dependency of all other SDKs concurrently. This helps ensure that users running multiple SDKs from a single machine all use the same engine. This is important as currently our automatic provisioning code enforces the existence of a single engine running at a time.
+
```

Comment:
**This is super important**. I thought that multiple engines with different versions are supported. Great to know this!

cc @aluzzardi @marcosnils

Reply:
Yes, more context in this comment and the one right above it: https://github.com/dagger/dagger/pull/3647#issuecomment-1302721770

tl;dr it's a simple solution for now, we can find ways of improving it in the near future (at the cost of some extra complexity)